projects/igniteui-angular/src/lib/grids/selection/selection.service.ts

```diff
@@ -53,20 +54,24 @@ export class IgxRow {
     state: any;
     newData: any;
 
-    constructor(public id: any, public index: number, public data: any) { }
+    constructor(public id: any, public index: number, public data: any, public grid: IgxGridBaseDirective & GridType) { }
 
-    createEditEventArgs(): IGridEditEventArgs {
-        return {
+    createEditEventArgs(includeNewValue = true): IGridEditEventArgs {
+        const args: IGridEditEventArgs = {
             rowID: this.id,
+            rowData:  { ... this.data },
```

Comment:
Might be good to leave the reference to the actual data record in case the event handler want to find in the data